File: tests/octave/test1.m
#!/usr/bin/env -S octave-cli -qf
% Nullity test of Divergence operator

addpath('/usr/share/mole/matlab/')

tol = 1e-10;

for k = [2, 4, 6, 8]; % Different orders of accuracy
    m = 2 * k + 1;
    dx = 1 / m;
    D = div(k, m, dx);
    field = ones(m + 1, 1);
    sol = D * field;

    if (norm(sol) > tol)
        fprintf("Test FAILED!\n");
    end

end

fprintf("Test PASSED!\n");

File: tests/cpp/test1.cpp
/**
 * Nullity test of Divergence operator
 */

#include <mole/divergence.h>
#include <mole/operators.h>
#include <gtest/gtest.h>

void run_nullity_test(int k, Real tol)
{
  int m = 2 * k + 1;
  Real dx = 1;

  Divergence D(k, m, dx);
  arma::vec field(m + 1, fill::ones);

  arma::vec sol = D * field;

  EXPECT_NEAR(arma::norm(sol), 0, tol);
}

TEST(DivergenceTests, Nullity)
{
  Real tol = 1e-10;
  for (int k : {2, 4, 6})
  {
    run_nullity_test(k, tol);
  }
}

int main(int argc, char **argv)
{
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}
